tests/chainer_tests/dataset_tests/test_convert.py

```diff
@@ -210,4 +210,45 @@ def test_concat_dicts_padding_gpu(self):
         self.check_concat_dicts_padding(cuda.cupy)
 
 
+class TestConcatExamplesWithBuiltInTypes(unittest.TestCase):
+
+    def get_int_arrays(self):
+        return [1, 2, 3]
+
+    def get_float_arrays(self):
+        return [1.0, 2.0, 3.0]
+
+    def check_device(self, array, device):
+        if device is not None:
+            self.assertIsInstance(array, cuda.ndarray)
+            self.assertEqual(array.device.id, device)
+
+    def check_concat_arrays(self, arrays, device=None, expected_type=None):
+        array = dataset.concat_examples(arrays, device)
+        self.assertEqual(array.shape, (len(arrays),))
+        self.check_device(array, device)
+
+        for x in array:
```

Comment:
I added value check using `numpy.testing.assert_array_equal`. Since the behavior of `cuda.to_cpu` is different for `numpy.int64` and `cupy.int64`, the test code became a little lengthy. Although I believe that we can do type-check and value-check, this code may be simplified.